examples/cpp/seerep_ros_communication/src/rosbagDumper.cpp

```diff
@@ -0,0 +1,370 @@
+#include "seerep_ros_communication/rosbagDumper.h"
+
+namespace seerep_grpc_ros
+{
+RosbagDumper::RosbagDumper(const std::string& bagPath, const std::string& classesMappingPath,
+                           const std::string& hdf5FilePath, const std::string& projectFrameId,
+                           const std::string& projectName, const std::string& topicImage,
+                           const std::string& topicCameraIntrinsics, const std::string& topicDetection,
+                           const std::string& detectionCategory, const std::string& topicTf,
+                           const std::string& topicTfStatic, const std::string& topicGeoAnchor,
+                           float distanceCameraGround, bool storeImages)
+  : hdf5FilePath(hdf5FilePath)
+  , projectFrameId(projectFrameId)
+  , projectName(projectName)
+  , topicImage(topicImage)
+  , topicCameraIntrinsics(topicCameraIntrinsics)
+  , topicDetection(topicDetection)
+  , detectionCategory(detectionCategory)
+  , topicTf(topicTf)
+  , topicTfStatic(topicTfStatic)
+  , topicGeoAnchor(topicGeoAnchor)
+  , distanceCameraGround(distanceCameraGround)
+{
+  auto writeMtx = std::make_shared<std::mutex>();
+  std::shared_ptr<HighFive::File> hdf5File =
+      std::make_shared<HighFive::File>(hdf5FilePath, HighFive::File::OpenOrCreate);
+  ioCoreGeneral = std::make_shared<seerep_hdf5_core::Hdf5CoreGeneral>(hdf5File, writeMtx);
+  ioTf = std::make_shared<seerep_hdf5_fb::Hdf5FbTf>(hdf5File, writeMtx);
+  ioImage = std::make_shared<seerep_hdf5_fb::Hdf5FbImage>(hdf5File, writeMtx);
+  ioImageCore = std::make_shared<seerep_hdf5_core::Hdf5CoreImage>(hdf5File, writeMtx);
+  ioPoint = std::make_shared<seerep_hdf5_fb::Hdf5FbPoint>(hdf5File, writeMtx);
+
+  ioCoreGeneral->writeProjectname(std::filesystem::path(bagPath).filename());
+  ioCoreGeneral->writeProjectFrameId("map");
+  ioCoreGeneral->writeVersion("0.0.0");
```

Comment:
Since the version number is now optional, I would suggest leaving the value empty if there is an appropriate value available. TLDR delete this line?